sematic/ui/src/pipelines/PipelineRunView.tsx

```diff
@@ -48,24 +48,6 @@ export default function PipelineRunView() {
     fetchResolution(rootId!);
   }, [rootId, fetchRootRun, fetchResolution]);
 
-  const changeRootRunId = useCallback(
-    (rootId: string) => {
-      if (rootRun && rootRun.id === rootId) return;
-      var runURL =
-        window.location.protocol +
-        "//" +
-        window.location.host +
-        "/pipelines/" +
-        calculatorPath +
-        "/" +
-        rootId;
-      window.history.pushState({ path: runURL }, "", runURL);
```

Comment:
If `React-router` is used in an application. `window.history.pushState()` must be avoided. Since [it is not monitored](https://reactrouter.com/en/main/start/concepts#history-object).

Using it might lead to many unnecessary state coalescing logics. Relying on the router's URL transition keeps everything clean and elegant.